File: editor/en/[1249]Minimum Remove to Make Valid Parentheses.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} s
 * @return {string}
 */
var minRemoveToMakeValid = function (s) {
    let stack = [];
    let sb = s.split('');

    for (let i = 0; i < s.length; i++) {
        if (sb[i] === '(') {
            stack.push(i);
        } else if (sb[i] === ')') {
            if (stack.length) {
                stack.pop();
            } else {
                sb[i] = '*';
            }
        }
    }

    while (stack.length) {
        sb[stack.pop()] = '*';
    }

    return sb.join('').replace(/\*/g, "");
};
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[1249]Minimum Remove to Make Valid Parentheses.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        stack = []
        sb = list(s)
        for i in range(len(sb)):
            if sb[i] == '(':
                stack.append(i)
            elif sb[i] == ')':
                if stack:
                    stack.pop()
                else:
                    sb[i] = '*'

        while stack:
            sb[stack.pop()] = '*'

        return ''.join(sb).replace('*', '')

# leetcode submit region end(Prohibit modification and deletion)
